code/workspaces/auth-service/src/permissions/permissions.js

```diff
@@ -31,36 +31,48 @@ const projectifyPermissions = ({ projectKey, permissions }) => {
   return permissions.map(permission => prefix.concat(permission));
 };
 
-//    [ { role: 'instanceAdmin', permissions: [/see permissions.yml/] } ]
-// -> [ ['system:instance:admin', ...] ]
-const systemifyPermissions = ({ permissions }) => {
-  const prefix = `${SYSTEM}${roleDelim}`;
-  return permissions.map(permission => prefix.concat(permission));
-};
-
 const flattenArray = (previous, current) => {
   if (Array.isArray(current)) {
     return [...previous, ...current];
   }
   return previous;
 };
 
+function getInstanceAdminPermissions(instanceRole) {
+  if (!instanceRole) {
+    return [];
+  }
+  const { permissions } = permissionAttributes[INSTANCE_ADMIN_ROLE_KEY];
+  return permissions.map(permission => `${SYSTEM}${roleDelim}${INSTANCE}${roleDelim}${permission}`);
```

Comment:
I find this a bit more readable. What do you think?

```js
return permissions.map(permission => [SYSTEM, INSTANCE, permission].join(roleDelim));
```

I think the same thing is used later on too. Might be worth splitting this into a separate function? e.g. 

```js
const constructPermission = (...permissionComponents) => permissionComponents.join(roleDelim);
```

which could then be used as 

```js
return permissions.map(permission => constructPermission(SYSTEM, INSTANCE, permission));
```

Reply:
Nice, done.